test/org/sagebionetworks/bridge/services/ParticipantOptionsServiceImplTest.java

```diff
@@ -1,82 +1,149 @@
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
 
-import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.sagebionetworks.bridge.TestUserAdminHelper;
-import org.sagebionetworks.bridge.TestUserAdminHelper.TestUser;
+
+import org.sagebionetworks.bridge.BridgeUtils;
 import org.sagebionetworks.bridge.dao.ParticipantOption;
 import org.sagebionetworks.bridge.dao.ParticipantOption.SharingScope;
-import org.sagebionetworks.bridge.dynamodb.OptionLookup;
+import org.sagebionetworks.bridge.dao.ParticipantOptionsDao;
+import org.sagebionetworks.bridge.dynamodb.DynamoStudy;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifier;
 import org.sagebionetworks.bridge.models.studies.Study;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
-@ContextConfiguration("classpath:test-context.xml")
-@RunWith(SpringJUnit4ClassRunner.class)
-public class ParticipantOptionsServiceImplTest {
+import com.google.common.collect.Sets;
 
-    @Resource
-    private ParticipantOptionsService optionsService;
-    
-    @Resource
-    private StudyService studyService;
-    
-    @Resource
-    private TestUserAdminHelper helper;
+public class ParticipantOptionsServiceImplTest {
     
-    private TestUser testUser;
+    private static final String HEALTH_CODE = "AAA";
     
-    private Study study;
+    private ParticipantOptionsServiceImpl service;
+    private ParticipantOptionsDao mockDao;
     
     @Before
     public void before() {
-        study = studyService.getStudy(TEST_STUDY_IDENTIFIER);
-        testUser = helper.createUser(ParticipantOptionsServiceImplTest.class);
+        service = new ParticipantOptionsServiceImpl();
+        mockDao = mock(ParticipantOptionsDao.class);
+        service.setParticipantOptionsDao(mockDao);
+        
+        Study study = new DynamoStudy();
+        study.setDataGroups(Sets.newHashSet("A","B","group1","group2","group3"));
     }
 
-    @After
-    public void after() {
-        if (testUser != null) {
-            helper.deleteUser(study, testUser.getEmail());
-        }
+    @Test
+    public void setBoolean() {
+        service.setBoolean(TEST_STUDY, HEALTH_CODE, EMAIL_NOTIFICATIONS, true);
+        
+        verify(mockDao).setOption(TEST_STUDY, HEALTH_CODE, ParticipantOption.EMAIL_NOTIFICATIONS, Boolean.TRUE.toString());
+        verifyNoMoreInteractions(mockDao);
     }
     
     @Test
-    public void canCrudSharingScopeOption() {
-        optionsService.setOption(testUser.getStudyIdentifier(), testUser.getUser().getHealthCode(),
-                SharingScope.ALL_QUALIFIED_RESEARCHERS);
+    public void getBoolean() {
+        when(mockDao.getOption(HEALTH_CODE, ParticipantOption.EMAIL_NOTIFICATIONS)).thenReturn("true");
```

Comment:
The service just checks for null parameters to the method and then calls to the DAO, and those null checks are trivial and not very interesting. You want those tested as well?